django/db/models/base.py

```diff
@@ -900,31 +900,42 @@ def save_base(
 
     save_base.alters_data = True
 
-    def _save_parents(self, cls, using, update_fields):
+    def _save_parents(self, cls, using, update_fields, updated_parents=None):
         """Save all the parents of cls using values from self."""
         meta = cls._meta
         inserted = False
+        if updated_parents is None:
+            updated_parents = {}
         for parent, field in meta.parents.items():
             # Make sure the link fields are synced between parent and self.
             if (
                 field
+                and hasattr(field, "attname")
                 and getattr(self, parent._meta.pk.attname) is None
                 and getattr(self, field.attname) is not None
             ):
                 setattr(self, parent._meta.pk.attname, getattr(self, field.attname))
-            parent_inserted = self._save_parents(
-                cls=parent, using=using, update_fields=update_fields
-            )
-            updated = self._save_table(
-                cls=parent,
-                using=using,
-                update_fields=update_fields,
-                force_insert=parent_inserted,
-            )
-            if not updated:
+
+            if (parent_updated := updated_parents.get(parent)) is None:
+                parent_inserted = self._save_parents(
+                    cls=parent,
+                    using=using,
+                    update_fields=update_fields,
+                    updated_parents=updated_parents,
+                )
+                updated = self._save_table(
+                    cls=parent,
+                    using=using,
+                    update_fields=update_fields,
+                    force_insert=parent_inserted,
+                )
+                if not updated:
+                    inserted = True
+                updated_parents[parent] = updated
+            elif not parent_updated:
                 inserted = True
             # Set the parent's PK value to self.
-            if field:
+            if field and hasattr(field, "attname"):
```

Comment:
Please move this fix & related tests to a separate PR. Also, I'm not sure if it is the right solution. Are we not just hiding the real issue?

Reply:
moved this fix to a new PR #16952 , also added some regression tests to confirm we are not hiding the real issue.